### PR TITLE
[R-package] added argument eval_train_metric to lgb.cv() (fixes #4911)

### LINKED ISSUE #4911: [R-package] add flag of displaying train loss for lgb.cv()
In https://github.com/microsoft/LightGBM/issues/1105, it was suggested to add the option `eval_train_metric` to `lgb.cv()` in order to see not only CV scores but also training scores. This is very useful to monitor overfitting. 

In Python, it was implemented in https://github.com/microsoft/LightGBM/pull/2089. 

I would love to see something similar for R.

### PR DESCRIPTION
**Pull request by mayer79**
Attempt to solve https://github.com/microsoft/LightGBM/issues/4911 

If we set `eval_train_metric = TRUE` in `lgb.cv()`, then we get output like 

[1] "[50]:  train's binary_logloss:0.248276+0.00109575  valid's binary_logloss:0.248655+0.00448549"

The resulting object "cvm" contains the information in the `record_evals` slot and the average training performance of the best round could be found by something like `cvm$record_evals$train$binary_logloss$eval[[cvm$best_iter]]`.

There is no unit test yet, but that does not mean we shouldn't write one.

### CODE REVIEW

File: R-package/R/lgb.cv.R [hunk 1]
```diff
@@ -78,6 +81,7 @@ lgb.cv <- function(params = list()
                    , record = TRUE
                    , eval_freq = 1L
                    , showsd = TRUE
+                   , eval_train_metric = FALSE
```
Comment (jameslamb) [inline R-package/R/lgb.cv.R:84]: This is a break change from the perspective of anyone calling `lgb.cv()` with positional arguments. That's probably unlikely for a function with this many arguments, but I still would like to make this change in a non-breaking way.

Please move `eval_train_metric` to the end of the function signature, after `serializable`.

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thanks for picking this up!

Please do add a unit test testing this behavior. Let me know if you need help with it.

Please re-generate the documentation.

```sh
sh build-cran-package.sh --no-build-vignettes
R CMD INSTALL --with-keep.source ./lightgbm_3.3.1.99.tar.gz
cd R-package
Rscript -e "roxygen2::roxygenize(load = 'installed')"
```

...to fix issues like the following in CI (e.g. [in this build](https://github.com/microsoft/LightGBM/runs/4651853462?check_suite_focus=true))

```text
Codoc mismatches from documentation object 'lgb.cv':
lgb.cv
  Code: function(params = list(), data, nrounds = 100L, nfold = 3L,
                 label = NULL, weight = NULL, obj = NULL, eval = NULL,
                 verbose = 1L, record = TRUE, eval_freq = 1L, showsd =
                 TRUE, eval_train_metric = FALSE, stratified = TRUE,
                 folds = NULL, init_model = NULL, colnames = NULL,
                 categorical_feature = NULL, early_stopping_rounds =
                 NULL, callbacks = list(), reset_data = FALSE,
                 serializable = TRUE)
  Docs: function(params = list(), data, nrounds = 100L, nfold = 3L,
                 label = NULL, weight = NULL, obj = NULL, eval = NULL,
                 verbose = 1L, record = TRUE, eval_freq = 1L, showsd =
                 TRUE, stratified = TRUE, folds = NULL, init_model =
                 NULL, colnames = NULL, categorical_feature = NULL,
                 early_stopping_rounds = NULL, callbacks = list(),
                 reset_data = FALSE, serializable = TRUE)
  Argument names in code not in docs:
    eval_train_metric
  Mismatches in argument names (first 3):
    Position: 13 Code: eval_train_metric Docs: stratified
    Position: 14 Code: stratified Docs: folds
    Position: 15 Code: folds Docs: init_model
```

File: R-package/R/lgb.cv.R (hunk 1 above)
Comment (mayer79) [reply R-package/R/lgb.cv.R:84]: Okay.

File: R-package/tests/testthat/test_basic.R [hunk 2]
```diff
@@ -554,6 +554,45 @@ test_that("lgb.cv() respects showsd argument", {
...
+    , showsd = FALSE
+    , eval_train_metric = FALSE
+  )
+  expect_equal(
+    bst_train$record_evals[["valid"]][["l2"]]
+    , bst_no_train$record_evals[["valid"]][["l2"]]
+  )
+  expect_true("train" %in% names(bst_train$record_evals))
+  expect_false("train" %in% names(bst_no_train$record_evals))
+  expect_is(bst_train$record_evals[["train"]][["l2"]][["eval"]], "list")
```
Comment (jameslamb) [inline R-package/tests/testthat/test_basic.R:589]:
```suggestion
  expect_true(methods::is(bst_train$record_evals[["train"]][["l2"]][["eval"]], "list"))
```

Could you remove this use of `testthat::expect_is()`? I recently learned that that function in the current `{testthat}` release series. See the description of #4916 for more details.

**Review by jameslamb — CHANGES_REQUESTED** [review]
thanks very much! The new test looks great! Just one more small suggestion.

**Review by jameslamb — APPROVED** [review]
looks great, thanks!

**Comment by jameslamb** [comment]
Thanks as always for the help @mayer79 !

**Comment by mayer79** [comment]
> Thanks as always for the help @mayer79 !

Thanks go to you, sir :-)
